smarts/core/scenario.py

```diff
@@ -317,61 +304,36 @@ def _discover_social_agents_info(
         if not os.path.exists(social_agents_path):
             return []
 
-        # [ ( missions_file, agent_actor, Mission ) ]
-        agent_bucketer = []
-
-        # like dict.setdefault
-        def setdefault(l: Sequence[Any], index: int, default):
-            while len(l) < index + 1:
-                l.append([])
-            return l[index]
-
         file_match = os.path.join(social_agents_path, "*.pkl")
+        items = []
         for missions_file_path in glob.glob(file_match):
             with open(missions_file_path, "rb") as missions_file:
-                count = 0
-                missions = pickle.load(missions_file)
-
-            for mission_and_actor in missions:
-                # Each pickle file will contain a list of actor/mission pairs. The pairs
-                # will most likely be generated in an M:N fashion
-                # (i.e. A1: M1, A1: M2, A2: M1, A2: M2). The desired behavior is to have
-                # a single pair per concrete Scenario (which would translate to
-                # "per episode" when swapping)
-                assert isinstance(
-                    mission_and_actor.actor, sstudio_types.SocialAgentActor
-                )
-
-                actor = mission_and_actor.actor
-                extracted_mission = Scenario._extract_mission(
-                    mission_and_actor.mission, road_map
-                )
-                namespace = os.path.basename(missions_file_path)
-                namespace = os.path.splitext(namespace)[0]
-
-                setdefault(agent_bucketer, count, []).append(
-                    (
-                        SocialAgent(
-                            id=SocialAgentId.new(actor.name, group=namespace),
-                            name=actor.name,
-                            is_boid=False,
-                            is_boid_keep_alive=False,
-                            agent_locator=actor.agent_locator,
-                            policy_kwargs=actor.policy_kwargs,
-                            initial_speed=actor.initial_speed,
-                        ),
-                        extracted_mission,
+                actors, missions = pickle.load(missions_file)
+            namespace = os.path.basename(missions_file_path)
+            namespace = os.path.splitext(namespace)[0]
+
+            social_agent_infos = [
+                (agent.id, agent)
+                for agent in (
+                    SocialAgent(
+                        id=SocialAgentId.new(actor.name, group=namespace),
```

Comment:
Is there a need to use a for loop here? Can't you just assign the `agent` variable the newly created `SocialAgent` instance?